### DISCUSSION TITLE
realtime-js - RealtimeSubscription connection error

### DISCUSSION THREAD

**Discussion by smarcet** [supabase/supabase#8669] [Questions]
hello guys 
i was looking at https://github.com/supabase/realtime-js/blob/master/src/RealtimeSubscription.ts source code and i have this doubt.
here
https://github.com/supabase/realtime-js/blob/10228a3d7cb87897751952dfd2f39d0a09f4cdac/src/RealtimeSubscription.ts#L49
when the realtime subscription got an error its trigger the reconnection timer, but timer callback is asking here
https://github.com/supabase/realtime-js/blob/10228a3d7cb87897751952dfd2f39d0a09f4cdac/src/RealtimeSubscription.ts#L64
for this.socket.isConnected(), what if the socket is not connected ? then the realtime subscription its left on a broken state ? ( like not receiving any more updates from server till subscription is restarted by client code ?)
much thanks in advance 
Seba

**Comment by GaryAustin1** [comment]
Yes realtime can/will lose the connection for various reasons.  You need to monitor the connection status and decide what to do if it disconnects and reconnects (like reload data as updates might have been missed).  There is no queue on the server side while the connection is gone.

I show one approach here in this discussion [How to obtain reliable realtime updates in the real world  #5641](https://github.com/supabase/supabase/discussions/5641)  and present some of the scenarios you can/will lose connection, including background tabs and devices sleeping.

Edit.  I may not be addressing your exact issue in re-reading it as you seem to believe there is an error in that code and it won't reconnect.  I have seen it reconnecting many times, but not sure if there is a specific case it could fail.   You could report an issue in realtime-js if you think there is a bug in the code.   My approach would hide such an error as I close the connection and restart it on my own.

**Reply by smarcet** [reply]
@GaryAustin1 thanks for your reply , and yes i do saw your discussion , was really helpful , much thanks!
but i left the tab opened like 2 hours or so, and from time to time, when i got on the subscribe callback a SUBSCRIPTION_ERROR
connection was not recovering gracefully and real time updates got broken, so in order to overcome that , on any SUBSCRIPTION_ERROR i opted to close the subscription and re open it, that approach allow me to leave the page opened for 24 hours and got all the updates without having a connection break
regards

**Reply by GaryAustin1** [reply]
Yeah, I just added to my reply as I would not see that error as I also just close and reconnect every time.
